lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -602,12 +602,36 @@ class ConvertAtenIndexTensorOp : public OpConversionPattern<AtenIndexTensorOp> {
       for (auto i : llvm::seq(startIndex, startIndex + maxRank)) {
         auto resultDimSize = refinedResultShape[i];
         if (ShapedType::isDynamic(resultDimSize)) {
-          return rewriter.notifyMatchFailure(
-              op, "unimplemented: index tensors must have static shape if "
-                  "there is more than one index tensor");
+          SmallVector<Value> dynamicDims;
+          int64_t staticDim = -1;
+          for (auto indexTensor : indexTensors) {
+            RankedTensorType indexTensorType =
+                indexTensor.getType().cast<RankedTensorType>();
+            int64_t indexTensorRank = indexTensorType.getRank();
+            if ((maxRank - indexTensorRank) > (i - startIndex))
+              continue;
+            int64_t dim = i - startIndex - maxRank + indexTensorRank;
+            if (ShapedType::isDynamic(indexTensorType.getShape()[dim]))
+              dynamicDims.push_back(getDimOp(rewriter, loc, indexTensor, dim));
+            else
+              staticDim = indexTensorType.getShape()[dim];
```

Comment:
I think there is one case missing here where there could be multiple static dims with one of them being 1 and the other being non-1. This should be the maximum between the current staticDim and new static dim values to handle this case. This also probably warrants an E2E test.